### PR TITLE
[docs] update header include coding guideline

### PR DESCRIPTION
**Pull request by Rechi**
## Motivation and Context
have a coding standard for includes which can also be checked by clang-format

## How Has This Been Tested?
run clang-format on the include blocks, see https://github.com/xbmc/xbmc/compare/master...Rechi:clang-format-includes for the result

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: docs/CODE_GUIDELINES.md [hunk 1]
```diff
@@ -366,43 +366,45 @@ static void test();
...
+Included header files have to be sorted (case sensitive) alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
 
 Header order has to be:
 * Own header file
-* Other Kodi includes
+* Other Kodi includes (platform independent)
+* Other Kodi includes (platform specific)
 * C and C++ system files
 * Other libraries' header files
+* special Kodi headers (PlatformDefs.h, system.h and system_gl.h)
```
Comment (ksooo) [inline docs/CODE_GUIDELINES.md]: Why are those "special" and why shall they placed after all other headers?

File: docs/CODE_GUIDELINES.md [hunk 2]
```diff
@@ -366,43 +366,45 @@ static void test();
...
 ## 7. Headers
-Included header files have to be sorted alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
+Included header files have to be sorted (case sensitive) alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
 
 Header order has to be:
 * Own header file
-* Other Kodi includes
+* Other Kodi includes (platform independent)
+* Other Kodi includes (platform specific)
 * C and C++ system files
```
Comment (ksooo) [inline docs/CODE_GUIDELINES.md]: C++ before C?

Comment (Rechi) [reply docs/CODE_GUIDELINES.md]: no in the same block

Comment (ksooo) [reply docs/CODE_GUIDELINES.md]: why?

Comment (Rechi) [reply docs/CODE_GUIDELINES.md]: it was like that before, if wanted it can be spilt into separate blocks

Comment (ksooo) [reply docs/CODE_GUIDELINES.md]: No hard feelings. Just wanted to know whether there is other reason than "was like that before".

File: docs/CODE_GUIDELINES.md (hunk 1 above)
Comment (Rechi) [reply docs/CODE_GUIDELINES.md]: PlatformDefs.h and system.h declares many windows like structs and defines which shouldn't pollute other headers if not needed
system_gl.h is a `Other libraries' header files` wrapper

Comment (ksooo) [reply docs/CODE_GUIDELINES.md]: okay, thanks. makes sense.

**Review by yol — APPROVED** [review]
Thanks!

File: docs/CODE_GUIDELINES.md (hunk 2 above)
Comment (lrusak) [reply docs/CODE_GUIDELINES.md]: I've been doing c++ before c but I'm indifferent

**Review by lrusak — APPROVED** [review]
It's nice to have this in writing. Thanks for doing it!

**Comment by Rechi** [comment]
@notspiff ping

**Review by notspiff — APPROVED** [review]

**Pull request merged**